### DISCUSSION TITLE
Questions on (multiple) forcing settings

### DISCUSSION THREAD

**Discussion by sclee1717** [NASA-LIS/LISF#861] [Q&A] [answered]
Hi all!

I'm trying to run Noah3.6 model with gdas and other precipitation forcings.
I got some questions on the met forcing settings.

1. When I want to restart the model with 2 met forcings with restart file that is created using only gdas ("Met forcing sources: gdas" in both ldt.config and lis.config), is it necessary to rebuild LSM parameter processing file with multiple met forcing sources option in LDT, or is it fine to progress with the same LSM parameter file? It seems there's no visible different result in the LDT-created parameter file selecting different met forcing sources; however, I checked that if I try to run LIS with "Number of surface model types: 1 (LSM)" option with parameter file that has been created with "Number of surface model types: 2 (LSM, Openwater)" option on LDT, it leads to an error.

2. I have checked multiple forcing-related issue at _https://modelingguru.nasa.gov/message/10519#10519_, and I found that the LIS UG says "_the forcing data specified in the second column will be used in place of forcing data that is specified in the first column, for locations within the spatial extent of the second column’s forcing data_." There only is an explanation about spatial extent of the forcing datasets, but no explanations related to "specific forcing variables". What could I expect if I wrote, for example, "Met forcing sources: gdas GPM IMERG"? Will only precipitation field from GPM replace precipitation from gdas and the other forcing variables from gdas will be used?

3. Which source code contains information on the "name" of the forcing variable LIS tries to find (which is specified in the forcing_variables.txt)? I'm trying to find out, for example:
 - is the precipitation in gdas counted as rainfall rate or convective rainfall rate?
 - which short name (or long name) does LIS try to find for rainfall rate in gdas?

Please kindly let me know how to solve the issues.
Hope this also helps someone having the same curiosity.

Thanks a lot!
Regards,
Lee

**Comment by karsenau** [comment, marked as answer]
Hi Lee (@sclee1717),

Thank you for your interest in using the LIS Framework (LISF) for your work. 
We will try to help answer your questions where we can. Please find some replies to your questions below.

1)  To run with 2 meteorological forcing datasets at the same time, where you "overlay" the forcing datasets in the same LIS land surface model (LSM) run, you will only need one LDT-processed input parameter file into LIS.  In terms of the model restart files, LIS will write out a single restart file at the output interval (e.g., daily or "1da") that you specify in the lis.config file, and is not dependent on the number of forcing data options that you chose.  

The only time this will change is if you are running LIS in what are known as "nests", which takes its history from running LIS with WRF (which is a more complex run mode and do not recommend for those first starting out to work with LIS). However, if you ever want to run LIS on different spatial resolution domains, which are referred to as nesting", it is possible, but you will then have to generate different input parameter files and different restart files will be written out for each nest.  This option is not impacted on how you select different forcing options.

For your "surface model" question, you will most likely be running LIS with a land surface model (LSM), which is 1 type of surface model, and which we recommend all new users to start with.  There are additional options to run with, like 'Openwater', which allows meteorological forcing datasets to be processed still over ocean points that are not masked out by the landwater mask parameter (generated in the LDT parameter preprocessing step). 

So if you want to run with "Openwater" as well, you will designate this option in your LDT config file as:

Number of surface model types:          2
Surface model types:                    "LSM"  "Openwater"

And in your lis.config file as:
Number of surface model types:          2
Surface model types:                    "LSM"  "Openwater"

This is done primarily in how the 1-d vector tile arrays are written and accounted for between the two LISF parts of the software. This is very important to keep them consistent between all components of LISF (including LVT).

2)  To run with 2 meteorological forcing datasets at the same time, where you "overlay" the forcing datasets, yes, you will have two columns as entries, one for each of the specified forcing datasets, and the second listed forcing option will be "overlaid" on the first one listed.

This was originally set up for when you have a coarser, global forcing dataset (e.g., MERRA2) and want to overlay a finer scale forcing dataset that is only available for a region, like NLDAS2 (e.g., continental U.S.). So MERRA2 would be specified for the first column, and then the NLDAS2 would be overlaid as specified in the second column so a user could have higher resolution info over the US. That combo would like, depending on your LIS domain grid target resolution, for example could be 0.15 deg:

Number of met forcing sources:          2
Met forcing sources:                          "MERRA2" "NLDAS2"
Blending method for forcings:          "overlay"
Topographic correction method (met forcing):  "none" "none"
Enable spatial downscaling of precipitation:   0   0
Spatial interpolation method (met forcing):   "budget-bilinear" "neighbor"  
Spatial upscaling method (met forcing):         "none"  "none"
Temporal interpolation method (met forcing):  "linear" "linear"

The other option that you are trying to set up and use, whereby replacing GDAS precipitation with a higher resolution observation-based precipitation product, is the other way in which we have implemented and supported this "overlay" option.

3) For your final question about the naming of the forcing datasets and what goes into your lis.config file, please check the following options in your lis/configs/lis.config.adoc file -- about lines 290-366, of that document.

The GDAS forcing option combined with GPM/IMERG precipitation would like:

Number of met forcing sources:          2
Met forcing sources:                          "GDAS" "GPM IMERG"
Blending method for forcings:          "overlay"

And yes, only the GPM IMERG precipitation should replace the GDAS precipitation field and not affect the other fields. That is taken care of by the way we apply ESMF in LIS.

The GDAS precipitation includes both the total precipitation and also a convective precip option, but you will have to specify that option in your forcing_variables.txt input table, to turn it on for reading in GDAS convective precip. When the IMERG observed precip is read in, it will only replace the total precipitation field of GDAS and not the convective one.


We hope the above information can help you with getting your LIS runs going, and please let us know if anything is unclear or have any additional questions.

**Reply by sclee1717** [reply]
Hi @karsenau!

Thank you for your detailed answers. Really helped me a lot, saved my time.
Everything is being clear, but I might have to ask question no. 3 again in detail.
Sorry for the unclear question at the first place.

Opening the forcing datasets with panoply, I see the datasets have short names and long names for variables, e.g.,
"Convective_Precipitation_rate_surface_*_Hour_Average"
"Precipitation_rate_surface_*_Hour_Average"
for convective precipitation and total precipitation.

Below is what I'm guessing,
When I turn on the option for "Rainf" in the forcing_variables.txt, the LIS will try to find the name "Precipitation_rate_surface_*_Hour_Average" in gdas.
When I turn on the option "Rainf" in the forcing_variables.txt and selecting IMERG as forcing in lis.config, LIS will try to find the name "precipitationCal" in IMERG.
they differ in names of the variables..

Is it possible to see which name of the variable (e.g., precipitation -> "precipitationCal") LIS tries to find in the forcing files, for selected forcing data?
I'm guessing maybe some of the source codes contain such information..
or is it not the way how LIS finds the forcing variable (by name)?

Thanks again for your support!
Regards,
Lee

**Reply by karsenau** [reply]
Hi @sclee1717,

No problem, and I'm glad some of the initial information was helpful. I'll try to address your GDAS file - variable naming question below.

If you are referring to the Grib-1 formatted GDAS files that are read into the LIS-based GDAS forcing reader, found in:

lis/metforcing/gdas directory

Then the Fortran file of interest is:  read_gdas.F90, and the internal subroutine named:  "retrieve_gdas_variables".

In that GDAS forcing routine, you will see how the Grib-1 formatted files and their content are read in.  Grib formatted files typically are associated with "tables" and are organized by variable indexed information, referred to as PDS (or Product Definition Section) information.  We use the "pds5" index value to identify the variable in the GDAS GRIB files and the same values in an ordered array in the Fortran routine to identify which parameters or forcing variables to grab and read in.  The order in which we align forcing variables in LIS are in the following order:

!  1. T2          Air Temperature [K] 
!  2. Q2          Specific humidity [kg kg-1]
!  3. SWDNB       Downward shortwave radiation [W m-2] 
!  4. LWDNB       Downward longwave radiation [W m-2] 
!  5. U10         10m U-dir wind vector [m s-1]
!  6. V10         10m V-dir wind vector [m s-1] 
!  7. PSFC        Surface pressure [Pa]
!  8. PREC_ACC_NC Precipitation [mm]

So make sure to check your GDAS Grib-1 formatted files and contents, which you can also use wgrib utility to look at the file header and content.

Please let us know if you have any other questions.

**Reply by karsenau** [reply]
To answer your other question about how the forcing_variables.txt relates to what forcing variables are read in from different sources (e.g., GDAS, IMERG precip), it is mainly a simple table to inform LIS which meteorological variables to expect for a given model or forcing dataset.  

The naming convention of the actual precipitation variable (or other metforce variables) is dependent on the dataset and unique to the met forcing reader in LIS specific to that dataset.

Best thing to check is in the metforc reader code itself, as I just shared how to do for GDAS variables, so you know which variables specifically are being read in.

We try our best to read in the exact metforce variables that follow the ALMA or other CF compliant guidelines, but what are the typical forcing fields expected by the models (e.g., 2m air temp, 10m winds).

**Reply by sclee1717** [reply]
Hi @karsenau!

Following your recommendations,
I looked into the reader source codes for several forcings and checked the subroutine that grabs variables.
I also found the pds numbers through wgrib, and understood how the variables are called.

Both of your answers guided me to see what I wanted to see.
Thanks a lot.
I found it very interesting.

Regards,
Lee
